### PR TITLE
feat: breakpoint upload via MinIO

### PR DESCRIPTION
**Pull request by dayuy**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
breakpoint upload via MinIO
#### Which issue(s) this PR fixes

Fixes #

### CODE REVIEW

File: graphql-server/go-server/minio/client.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package minio
```
Comment (0xff-dev) [inline graphql-server/go-server/minio/client.go:1]: add license.

File: graphql-server/go-server/minio/client.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+var mutex *sync.Mutex
+
+var (
+	minioAddress         = "kubeagi-minio.172.22.96.136.nip.io"
+	minioAccessKeyId     = "admin"
+	minioSecretAccessKey = "Passw0rd!"
+	minioSecure          = false
+	minioBucket          = "arcadia"
+	minioBasePath        = "datasource"
```
Comment (dayuy) [inline graphql-server/go-server/minio/client.go:21]: ENV ?

Comment (0xff-dev) [reply graphql-server/go-server/minio/client.go:21]: Currently arcadia deployed via helm will have a configmap named `<release-name>-config`, which records information about the configuration Datasource for recording minio.

Here it is recommended to configure the flag parameter to support user configuration, if it is not set, just read it from the corresponding `configmap`.

Comment (Abirdcfly) [reply graphql-server/go-server/minio/client.go:21]: see pkg/config to find some helper functions

**Comment by Abirdcfly** [comment]
`go mod tidy` and `make prepare-push`

**Comment by bjwswang** [comment]
@dayuy Please make sure this works well when using webui to upload files.

File: graphql-server/go-server/main.go [hunk 3]
```diff
@@ -85,6 +86,12 @@ func main() {
 		http.Handle("/bff", srv)
 	}
 
+	http.HandleFunc("/minio/get_chunks", minio.GetSuccessChunks)
```
Comment (bjwswang) [inline graphql-server/go-server/main.go:89]: We need to do permission check in namespaced scope.

File: graphql-server/go-server/minio/client.go (hunk 2 above)
Comment (bjwswang) [reply graphql-server/go-server/minio/client.go:21]: @dayuy see @Abirdcfly

File: graphql-server/go-server/minio/client.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
+package minio
+
+import (
```
Comment (bjwswang) [inline graphql-server/go-server/minio/client.go:3]: @dayuy  move this `minio` under `go-server/pkg`

**Review by bjwswang — CHANGES_REQUESTED** [review]
@dayuy  Can you provide a common library to provde this file uplaod capability ?

Then we can discuss how to provide apis:

- upload file for a versioneddataset

File: graphql-server/go-server/minio/service.go [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+		if err != nil {
+			klog.Infof("GetFileChunkByMD5 failed: %s", err)
+			break
+		}
+		uuid = fileChunk.UUID
+		uploaded = strconv.Itoa(fileChunk.IsUploaded)
+		uploadID = fileChunk.UploadID
+		fileName = fileChunk.FileName
+
+		bucketName := minioBucket
```
Comment (bjwswang) [inline graphql-server/go-server/minio/service.go:86]: For versioneddataset, the bucketname should be namespace. Not hardcoded

File: graphql-server/go-server/main.go (hunk 3 above)
Comment (dayuy) [reply graphql-server/go-server/main.go:89]: TODO

File: graphql-server/go-server/minio/service.go (hunk 5 above)
Comment (dayuy) [reply graphql-server/go-server/minio/service.go]: TODO

**Review by bjwswang — APPROVED** [review]

**Pull request merged**